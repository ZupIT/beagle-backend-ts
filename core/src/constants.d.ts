export declare const coreNamespace = "beagle";
export declare const genericNamespace = "custom";
