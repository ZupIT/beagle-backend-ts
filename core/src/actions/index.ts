export { addChildren } from './add-children'
export { alert } from './alert'
export { condition } from './condition'
export { confirm } from './confirm'
export * from './navigation'
export { sendRequest, SendRequestParams } from './send-request'
export { setContext } from './set-context'
export { submitForm } from './submit-form'
