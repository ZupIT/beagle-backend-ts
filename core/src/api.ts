/* This file is only used for exporting the API documentation. We do this so we can have a full documentation without
exporting types that doesn't make sense. */

/**
 * @module @zup-it/beagle-backend-core
 */

export * from './'
export { Action, AnalyticsConfig, AnalyticsConfigObject, AnalyticsAttributesMap } from './model/action'
export { ContextNode } from './model/context/context-node'
