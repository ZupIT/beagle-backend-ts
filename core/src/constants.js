"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.genericNamespace = exports.coreNamespace = void 0;
exports.coreNamespace = 'beagle';
exports.genericNamespace = 'custom';
