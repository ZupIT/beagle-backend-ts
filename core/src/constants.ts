/**
 * The namespace used by the core Beagle actions and the components in the package "@zup-it/beagle-backend-components".
 */
export const coreNamespace = 'beagle'

/**
 * The default namespace when no namespace is provided to an action or component.
 */
export const genericNamespace = 'custom'

/**
 * String expected by the hot reloading service (cli: start) for identifying a server restart.
 *
 * @privateRemarks this string must the same as the constant of same name declared in the module "cli".
 */
export const hotReloadingString = '__[HOT RELOADING: SERVER_STARTED]__'
