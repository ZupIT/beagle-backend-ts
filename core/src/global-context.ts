import { createContextNode } from './model/context/context-node'

/**
 * The Global Context is a Beagle Context that doesn't need to be declared and is available for the entire lifetime of
 * the application, i.e. if you set the a value in teh global context in Screen A, the value will be available in Screen
 * B. Furthermore, the values in the global context can also be set by the frontend application.
 *
 * We advise the developer to define the structure of the global context as an interface, so it can be safely accessed.
 * When retrieving the global context, use `getGlobalContext<T>()`, where T is the interface.
 *
 * @returns a reference to the Global Context.
 */
export const getGlobalContext = <T>() => createContextNode<T>('global')
