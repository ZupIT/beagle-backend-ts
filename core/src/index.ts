export { Component, WithContext, WithChildren } from './model/component'
export { Expression, HttpMethod, DynamicExpression } from './types'
export { RootContext } from './model/context/root-context'
export { coreNamespace } from './constants'
export { Actions, WithAnalytics, createAction } from './model/action'
export { AnyContextNode } from './model/context/types'
export { ContextNode } from './model/context/context-node'
export { createContext } from './model/context'
export { serialize } from './serializer'
export { FC, ComponentProps } from './jsx/types'
export { logger } from './logger'
export { isDynamicExpression } from './utils'
export { BeagleJSX } from './jsx'

import { Component } from './model/component'


declare global {
  // eslint-disable-next-line @typescript-eslint/no-namespace
  namespace JSX {
    type Element = Component

    interface ElementChildrenAttribute {
      children: 'children',
    }

    interface IntrinsicElements {
      component: Element,
    }
  }
}
