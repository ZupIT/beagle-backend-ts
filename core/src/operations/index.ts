export * from './array'
export * from './number'
export * from './logic'
export * from './other'
