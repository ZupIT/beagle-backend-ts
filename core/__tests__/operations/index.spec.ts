import { mapValues } from 'lodash'
import * as operations from 'src/operations'

describe('Operations module', () => {
  it('should export', () => expect(mapValues(operations, v => typeof v)).toMatchSnapshot())
})
