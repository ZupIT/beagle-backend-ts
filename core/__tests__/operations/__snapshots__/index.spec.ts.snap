// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Operations module should export 1`] = `
Object {
  "and": "function",
  "condition": "function",
  "contains": "function",
  "divide": "function",
  "gt": "function",
  "gte": "function",
  "insert": "function",
  "isEmpty": "function",
  "isNull": "function",
  "length": "function",
  "lt": "function",
  "lte": "function",
  "multiply": "function",
  "not": "function",
  "or": "function",
  "remove": "function",
  "removeIndex": "function",
  "subtract": "function",
  "sum": "function",
  "union": "function",
}
`;
